features/step_definitions/buyer_journey.rb

```diff
@@ -0,0 +1,82 @@
+  service_status_public("1123456789012346")
+  service_status_public("1123456789012347")
+  service_status_public("1123456789012348")
+  service_status_public("1123456789012349")
+  service_status_public("1123456789012350")
+  service_status_public("1123456789012351")
+  service_status_public("1123456789012352")
+  service_status_public("1123456789012353")
+  service_status_public("1123456789012354")
+end
+
+Given(/I navigate to the list of '(.*)' page$/) do |value|
+  page.visit("#{dm_frontend_domain}/g-cloud/suppliers?prefix=#{value.split(' ').last}")
+  @data_store = @data_store || Hash.new
+  @data_store['supplier_alphabet'] = "#{value.split(' ').last}"
+end
+
+Then(/I am taken to page '(.*)' of results$/) do |page_number|
+  if current_url.include?('search?')
+    current_url.should include("#{dm_frontend_domain}/g-cloud/search?")
+    current_url.should include("page=#{page_number}")
+    current_url.should include("lot=iaas")
+    if page_number >= '2'
+      page.should have_selector(:xpath, "//a[contains(text(), 'Next')]//following-sibling::span[contains(text(),'page')]")
+      page.should have_selector(:xpath, "//a[contains(text(), 'Next')]//following-sibling::span[contains(text(),'page')]/..//following-sibling::span[@class='page-numbers'][contains(text(), '3 of #{@data_store['numberofpages']}')]")
+      page.should have_selector(:xpath, "//a[contains(text(), 'Previous')]//following-sibling::span[contains(text(),'page')]")
+      page.should have_selector(:xpath, "//a[contains(text(), 'Previous')]//following-sibling::span[contains(text(),'page')]/..//following-sibling::span[@class='page-numbers'][contains(text(), '1 of #{@data_store['numberofpages']}')]")
+    elsif page_number < '2'
+      page.should have_selector(:xpath, "//a[contains(text(), 'Next')]//following-sibling::span[contains(text(),'page')]")
+      page.should have_no_selector(:xpath, "//a[contains(text(), 'Previous')]//following-sibling::span[contains(text(),'page')]")
+    end
+  elsif current_url.include?('suppliers?')
+    current_url.should include("#{dm_frontend_domain}/g-cloud/suppliers?")
+    current_url.should include("prefix=#{@data_store['supplier_alphabet']}")
+    current_url.should include("page=#{page_number}")
+    current_url.should include("framework=g-cloud")
+    if page_number >= '2'
+      page.should have_selector(:xpath, "//a[contains(text(), 'Previous')]//following-sibling::span[contains(text(),'page')]")
+    elsif page_number < '2'
+      page.should have_selector(:xpath, "//a[contains(text(), 'Next')]//following-sibling::span[contains(text(),'page')]")
+      page.should have_no_selector(:xpath, "//a[contains(text(), 'Previous')]//following-sibling::span[contains(text(),'page')]")
+    end
+  end
+end
+
+
+Then(/Pagination is '(.*)'$/) do |availability|
+  if current_url.include?('suppliers?')
+    pagination_available = "#{availability}"
+  else
+    fail("Pagination check only works for the suppliers list page these days.")
```

Comment:
I'm not sure this is true. We have pagination on [the dos opportunities page for example](https://www.digitalmarketplace.service.gov.uk/digital-outcomes-and-specialists/opportunities). Maybe this step could be moved to a common place and then used elsewhere.